packages/faceted-search/src/components/Badges/BadgeFaceted/BadgeFaceted.component.js

```diff
@@ -0,0 +1,133 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import Badge from '@talend/react-components/lib/Badge';
+import BadgeComposition from '@talend/react-components/lib/Badge/BadgeComposition';
+import { getTheme } from '@talend/react-components/lib/theme';
+
+import { BadgeOperatorOverlay } from '../BadgeOperator';
+import { BadgeOverlay } from '../BadgeOverlay';
+import { useBadgeFacetedContext } from '../../context/badgeFaceted.context';
+
+import cssModule from './BadgeFaceted.scss';
+
+import { useBadgeOverlayFlow, OVERLAY_FLOW_ACTIONS } from '../../../hooks/badgeOverlayFlow.hook';
+import { BADGES_ACTIONS } from '../../../hooks/facetedBadges.hook';
+
+import { operatorPropTypes, operatorsPropTypes } from '../../facetedSearch.propTypes';
+
+const theme = getTheme(cssModule);
+
+const findOperatorByName = name => operator => name === operator.name;
+
+const BadgeFaceted = ({
+	badgeId,
+	children,
+	id,
+	labelCategory,
+	labelValue,
+	initialOperatorOpened,
+	initialValueOpened,
+	operator,
+	operators,
+	value,
+	size = Badge.SIZES.large,
+	t,
+}) => {
+	const [
+		overlayState,
+		overlayDispatch,
+		onChangeOperatorOverlay,
+		onChangeValueOverlay,
+	] = useBadgeOverlayFlow(initialOperatorOpened, initialValueOpened);
+
+	const { state, dispatch, onSubmit } = useBadgeFacetedContext();
+
+	const [badgeOperator, setBadgeOperator] = useState(operator);
+	const [badgeValue, setBadgeValue] = useState(value);
+
+	const onChangeOperator = (_, operatorName) => {
+		const foundOperator = operators.find(findOperatorByName(operatorName));
+		if (foundOperator) {
+			setBadgeOperator(foundOperator);
+		}
+		overlayDispatch(OVERLAY_FLOW_ACTIONS.openValue);
+		dispatch(BADGES_ACTIONS.closeInitialOpened(badgeId));
+	};
+
+	const onChangeValue = (_, entityValue) => {
+		setBadgeValue(entityValue);
+	};
+
+	const onSubmitBadge = event => {
+		overlayDispatch(OVERLAY_FLOW_ACTIONS.closeAll);
+		event.preventDefault();
+		dispatch(
+			BADGES_ACTIONS.update(badgeId, {
+				value: badgeValue,
+				operator: badgeOperator,
+			}),
+		);
+		onSubmit(event, state.badges);
```

Comment:
Think i find a bug, here is the pattern:
- Start storybook, story FacetedSearch > default
- Add a text badge with value `TEST1` and apply
- On Actions log, you see a `onSubmit` event, but there is no `value` key on `properties`
- Click to edit to badge and change value to `TEST2`, then apply
- On Actions log, you see a `onSubmit` event, but there is no `value` key on `properties`

In my opinion, you use `state.badges` as args here, but is not already "muted" with your value. You need to compute an object here to be sure that your arg state is the good one